MAINTAINERS.md

```diff
@@ -12,7 +12,6 @@ Active Maintainers
 * Daniel Wagner-Hall
 * Eric Arellano
 * Henry Fuller
-* John Sirois
```

Comment:
I trust a maintainer or owner will remove me from all relevant Pantsbuild GitHub groups / perms / etc. I've completed transferring out the Pex repos and Pex is all running green after that migration.